packages/app/src/app/overmind/actions.ts

```diff
@@ -94,15 +94,32 @@ export const modalClosed: Action = ({ state }) => {
   state.currentModal = null;
 };
 
-export const signInClicked: AsyncAction<{ useExtraScopes: boolean }> = (
-  { actions },
-  options
-) => actions.internal.signIn(options);
+export const signInClicked: Action<string | void> = ({ state }, redirectTo) => {
+  state.signInModalOpen = true;
+  state.redirectOnLogin = redirectTo || '';
+};
+
+export const toggleSignInModal: Action = ({ state }) => {
+  state.signInModalOpen = !state.signInModalOpen;
+};
+
+export const signInButtonClicked: AsyncAction<{
+  useExtraScopes: boolean;
+}> = async ({ actions, state }, options) => {
+  if (!options) {
```

Comment:
Doesn't this mean `signInButtonClicked` should be `AsyncAction<{ useExtraScopes: boolean } | void>`? 🤔

Reply:
It does 😄

Reply:
Done :)